packages/core/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,239 @@
+import type { CSSProperties } from 'react';
+import { type ComponentPropsWithoutRef, forwardRef, useState } from 'react';
+
+import { Popover as BasePopover } from '@base-ui-components/react/popover';
+import clsx from 'clsx';
+
+import { useMutationObserver } from '~/hooks/use-mutation-observer';
+import { createContext } from '~/libs/create-context';
+import { vars } from '~/styles/vars.css';
+import { type PositionerProps, splitPositionerProps } from '~/utils/split-positioner-props';
+
+import * as styles from './popover.css';
+
+type SharedProps = PositionerProps;
+type ContextProps = SharedProps;
```

Comment:
Following the menu component convention, what do you think about changing it to match the menu component?
```suggestion
type PopoverSharedProps = PositionerProps;
type PopoverContextProps = PopoverSharedProps;
```